arctic/chunkstore/chunkstore.py

```diff
@@ -90,6 +90,7 @@ def _check_invalid_segment(self):
 
     @mongo_retry
     def _reset(self):
+        self._arctic_lib.reset_auth()
```

Comment:
I think what's a little bit awkward is requiring every implementation to `_reset_auth` on the LibraryBinding - rather than have the top-level Arctic object poke the bindings directly?  Otherwise we risk adding a new library type and forgetting to implement this reset functionality?

Reply:
unfortunatelly the Store implementations do not have a parent class so it is not possible to e.g. call super _reset() or do something similar.

The _reset() is being called only by Arcitc's reset() and form the init() of each Store implementation.

The auth details are stored/handled in ArcticLibraryBinding, so I could potentially:
 1. always trigger auth when get_top_level_collection() is called for first time 
 2. trigger auth in ArcticLibraryBinding whenever the underlying connection of Arctic has changed, i.e. Arctic's reset() has been called.

I'll push quickly a fix for this.

Reply:
The change is now pushed, please have a look

Reply:
added test for this exact scenario